src/library/scala/collection/Iterator.scala

```diff
@@ -212,66 +211,55 @@ trait Iterator[+A] extends IterableOnce[A] with IterableOnceOps[A, Iterator, Ite
       buf
     }
 
-    private def padding(x: Int) = immutable.ArraySeq.untagged.fill(x)(pad.get())
     private def gap = (step - size) max 0
 
     private def go(count: Int) = {
-      val prevSize = buffer.size
+      val prevSize = group.size
       def isFirst = prevSize == 0
+      val extension = takeDestructively(count)
       // If there is padding defined we insert it immediately
       // so the rest of the code can be oblivious
-      val xs: Seq[B] = {
-        val res = takeDestructively(count)
-        // was: extra checks so we don't calculate length unless there's reason
-        // but since we took the group eagerly, just use the fast length
-        val shortBy = count - res.length
-        if (shortBy > 0 && pad.isDefined) res ++ padding(shortBy) else res
+      var shortBy = count - extension.size
+      if (pad != null) while (shortBy > 0) {
```

Comment:
This is probably better than extra `++` (I wonder if these were iterators in an early version of the code), but I wonder no one asked about this indentation. I wonder if jit would optimize `while (pad != null && shortBy > 0)`? Not recommending. But for the amount of time it takes to grok this class, an extra NL is such a luxury.

Reply:
Also surprised old paulp expression was not `pad.foreach`.